linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -989,7 +993,9 @@ private final class ClassDefChecker(classDef: ClassDef,
         checkTrees(captureValues, env)
 
       case Transient(transient) =>
-        checkAllowTransients()
+        if (!featureSet.supports(FeatureSet.ofTransient(transient)))
```

Comment:
IIUC this will become moot with the new design of of desugaring. Otherwise, we might want to consider inverting definition here to be more consistent with the transient design (i.e. put the phase / feature set into the transient, not the other way around).